semver/mod.ts

```diff
@@ -431,351 +439,94 @@ for (let i = 0; i < R; i++) {
  * Attempt to parse a string as a semantic version, returning either a `SemVer`
  * object or `null`.
  */
-export function parse(
-  version: string | SemVer | null,
-  options?: Options,
-): SemVer | null {
-  if (typeof options !== "object") {
-    options = {
-      includePrerelease: false,
-    };
-  }
-
-  if (version instanceof SemVer) {
-    return version;
-  }
-
+export function parse(version: string | SemVer): SemVer {
```

Comment:
I think `parse` should only accept a strings,

Reply:
Fair enough, that would be cleaner.

Reply:
I refactored the whole thing such that the `parse()`, `parseComparator()` and `parseRange()` functions all only take strings now, and all of the other functions take either a `SemVer`, `Comparator` or `Range` argument specifically.

Its hard to understate how much complexity it adds to these apis to have every function attempting to check all possible types.

Reply:
@justinmchase Did you decide against the string-only argument — or just haven't modified that part yet?

https://github.com/denoland/deno_std/pull/3169/files#diff-581da22f7b2e7456281ca11fb4976d2ab48b8f5e89063da36861efe16e1e2a66R182

https://github.com/denoland/deno_std/blob/d5feeab5cae4f49b709d05d2e60ecfe276d416ad/semver/parse.ts#L182